### PR TITLE
Refactor existing Regex scalars to use Regex Base Type

### PR DESCRIPTION
**Pull request by wonbyte**
Refactor existing Regex scalars in `Hotchocolate.Types.Scalars` to inherit from the `RegexType`

Addresses #3175

### CODE REVIEW

File: global.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "sdk": {
     "version": "5.0.100"
-    
+    "rollForward": "latestMinor"
```
Comment (wonbyte) [inline global.json:4]: @michaelstaib should I leave this?

**Review by PascalSenn — DISMISSED** [review]

File: src/HotChocolate/Core/src/Types.Scalars/PostalCodeType.cs [hunk 2]
```diff
@@ -144,5 +144,46 @@ private static bool ValidatePostCode(string postCode)
 
             return false;
         }
+
+        private static class PostalCodePatterns
```
Comment (wonbyte) [inline src/HotChocolate/Core/src/Types.Scalars/PostalCodeType.cs:148]: 👍

**Pull request merged**